cf-log/cf-log-stack.ts

```diff
@@ -0,0 +1,274 @@
+                      {"name": "c_port", "type": "integer"},
+                      {"name": "time_to_first_byte", "type": "real"},
+                      {"name": "x_edge_detailed_result_type", "type": "varchar"},
+                      {"name": "sc_content_type", "type": "varchar"},
+                      {"name": "sc_content_len", "type": "bigint"},
+                      {"name": "sc_range_start", "type": "bigint"},
+                      {"name": "sc_range_end", "type": "bigint"},
+                      {"name": "year", "type": "varchar"},
+                      {"name": "month", "type": "varchar"},
+                      {"name": "day", "type": "varchar"},
+                      {"name": "hour", "type": "varchar"},
+                      {"name": "file", "type": "varchar"}
+                    ]
+                  }`), ' */')
+            }
+        })
+
+        const p0 = new iam.PolicyStatement({
+            effect: iam.Effect.ALLOW,
+            actions: ["athena:StartQueryExecution", "athena:GetQueryExecution"],
+            resources: ['*']
+        });
+        const p1 = new iam.PolicyStatement({
+            effect: iam.Effect.ALLOW,
+            actions: ["s3:ListBucket", "s3:GetBucketLocation"],
+            resources: [CloudFrontAccessLogsBucket.bucketArn]
+        });
+        const p2 = new iam.PolicyStatement({
+            effect: iam.Effect.ALLOW,
+            actions: ["s3:PutObject", "s3:GetObject"],
+            resources: [CloudFrontAccessLogsBucket.arnForObjects("/*")]
+        });
+        const p3 = new iam.PolicyStatement({
+            effect: iam.Effect.ALLOW,
+            actions: [
+                "glue:CreatePartition", "glue:GetDatabase", "glue:GetTable",
+                "glue:BatchCreatePartition", "glue:GetPartition", "glue:GetPartitions",
+                "glue:CreateTable", "glue:DeleteTable", "glue:DeletePartition"
+            ],
+            resources: ['*']
+        });
+
+        const TransformPartFn = new lambda.Function(this, 'TransformPartFn', {
+            runtime: lambda.Runtime.NODEJS_12_X,
+            code: lambda.Code.asset(`${__dirname}/functions`),
+            handler: 'transformPartition.handler',
+            timeout: cdk.Duration.seconds(900),
+            initialPolicy: [
+                p0, p1, p2, p3
+            ],
+            environment: {
+                SOURCE_TABLE: PartitionedGzTable.ref,
+                TARGET_TABLE: PartitionedParquetTable.ref,
+                DATABASE: CFLogsDatabase.ref,
+                ATHENA_QUERY_RESULTS_LOCATION: CloudFrontAccessLogsBucket.s3UrlForObject("athena-query-results")
+            }
+        });
+        const HourlyEvt0 = new events.Rule(this, 'HourlyEvt0', {
+            schedule: events.Schedule.expression('cron(1 * * * ? *)')
+        });
+        HourlyEvt0.addTarget(new targets.LambdaFunction(TransformPartFn));
+
+        const CreatePartFn = new lambda.Function(this, 'CreatePartFn', {
```

Comment:
Let's add metrics and alarms for those Lambda functions.

In case we're notified when something wrong happens.